react/features/welcome/components/WelcomePage.native.js

```diff
@@ -127,6 +123,13 @@ class WelcomePage extends AbstractWelcomePage<*> {
             headerRight: () =>
                 <VideoSwitch />
         });
+
+        navigation.addListener('focus', () => {
```

Comment:
Don't we want to do this in the `blur` event instead? Also, set the insecure warning to false too. I guess we should also stop the animation timers and re-arm them on focus.

Reply:
I'll look into it.  If I am correct, state is updated with the new room name when we press join, therefore Welcome screen is blurred. In that order, when we get back to the Welcome screen aka is focused, we want to clear the state.

Reply:
It is also focused the first time it's rendered, and we don't want to clean the state in that case, do we? It's innocuous, sure but still.

In general i think it makes more sense to do it on blur because it's more connected to the idea behind it: "clean the state and stop generating room names when the screen is not visible".